### PR TITLE
Added MmSupervisorCoreGuid 

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Added gMmSupervisorCoreGuid to the supervisor dec file.  The value of this GUID is exactly the same as the MmSupervisorCore.inf file GUID and it's used to reference supervisor module directly instead of relying on gEfiCallerGuid.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested with CI and a physical platform.  No issues observed.

## Integration Instructions

N/A

### CODE REVIEW

File: MmSupervisorPkg/MmSupervisorPkg.dec [hunk 1]
```diff
@@ -37,6 +37,7 @@
   gMmProtectedRegionHobGuid                       = { 0x6c0792ac, 0x13d7, 0x431b, { 0xa4, 0x89, 0x3, 0x2f, 0x4a, 0xf9, 0x73, 0x80 } }
   gMmSupervisorPolicyFileGuid                     = { 0x81ff0793, 0x3e18, 0x489b, { 0x9a, 0x8, 0xa1, 0xeb, 0x71, 0xb2, 0x3d, 0x20 } }
   gMmSupervisorDriverDispatchGuid                 = { 0x2e135da6, 0xade0, 0x4b96, { 0x9b, 0x40, 0xb2, 0xf3, 0x67, 0xe9, 0xf7, 0xf7 } }
+  gMmSupervisorCoreGuid                           = { 0x4e4c89dc, 0xa452, 0x4b6b, { 0xb1, 0x83, 0xf1, 0x6a, 0x2a, 0x22, 0x37, 0x33 } }
```
Comment (kuqin12) [inline MmSupervisorPkg/MmSupervisorPkg.dec:40]: Please add a comment to describe the value of this GUID

**Review by kuqin12 — APPROVED** [review]

File: MmSupervisorPkg/MmSupervisorPkg.dec (hunk 1 above)
Comment (kenlautner) [reply MmSupervisorPkg/MmSupervisorPkg.dec:41]: Updated.

**Comment by kuqin12** [comment]
Looks like you need to grant GUID duplication exception in the package dec.

**Review by apop5 — APPROVED** [review]

**Comment by kuqin12** [comment]
Test

**Pull request merged**